src/librustc/ty/instance.rs

```diff
@@ -45,6 +45,51 @@ pub enum InstanceDef<'tcx> {
     CloneShim(DefId, Ty<'tcx>),
 }
 
+impl<'a, 'tcx> TyCtxt<'a, 'tcx, 'tcx> {
+    pub fn is_inline_instance(
```

Comment:
Where is this used? In general, I'd expect these methods to be on `Instance`, or where possible, `InstanceDef`, and private if only used locally. (I was trying to come up with better name for them)

Reply:
This is only used in `requests_inline`afaik. But the original function is used in callee.rs and trans_item_rs. I can move it to instance, I wasn't sure where to put it because it takes an instance and a tyctxt.